<?xml version="1.0" encoding="ISO-8859-15"?>
<launch>
  <arg name="voxel_size" default="0.42"/>
  <arg name="num_agents"/>
  
  <!-- Voxblox Server Node -->
  <node name="voxblox_node" pkg="voxblox_ros" type="esdf_pose_graph_server" output="screen" args="-alsologtostderr" clear_params="true">
    <!-- Topics -->    
    <remap from="voxblox_node/esdf_map_out" to="esdf_map" />
    <remap from="voxblox_node/tsdf_map_out" to="tsdf_map" />

    <remap from="pointcloud_transform0" to="/pose_graph_backend/pcl_transform0"/>
    <remap from="pointcloud_transform1" to="/pose_graph_backend/pcl_transform1"/>
    <remap from="pointcloud_transform2" to="/pose_graph_backend/pcl_transform2"/>

    <param name="use_tf_transforms" value="true"/>
    <param name="world_frame"       value="world"/>
    <param name="num_agents"        value="$(arg num_agents)"/>

    <param name="use_freespace_pointcloud" value="false" />
    <param name="tsdf_voxel_size"          value="$(arg voxel_size)" />
    <param name="tsdf_voxels_per_side"     value="16" />
    <param name="voxel_carving_enabled"    value="true" />
    <param name="color_mode"               value="colors" />
    <param name="verbose"                  value="true" />
    <param name="update_mesh_every_n_sec"  value="1.0" />
    <param name="slice_level"              value="-0.5" />
    <param name="method"                   value="fast" />
    <param name="use_const_weight"         value="false" />
    <param name="publish_slices"           value="true" />
    <param name="publish_pointclouds"      value="true" />
    <param name="max_ray_length_m"         value="7"/>
    <param name="publish_traversable"      value="false"/>
    <param name="pointcloud_queue_size"    value="20"/>
    <param name="min_ray_length_m"         value="0.15"/>

    <rosparam file="$(find voxblox_ros)/cfg/ncamera_23026.yaml"/> 
  </node>
</launch>
